experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+        static Dictionary<string, string> Capitals = new Dictionary<string, string>
+        {
+            ["France"] = "Paris",
+            ["Italy"] = "Rome",
+            ["Japan"] = "Tokyo",
+            ["Poland"] = "Warsaw",
+            ["Germany"] = "Hamburg" // not really
+        };
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+            var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+            var userText = turnContext.Activity.Text.ToLower();
+            if (userText.Contains("agent"))
+            {
+                await turnContext.SendActivityAsync("Your request will be escalated to a human agent");
+
+                var transcript = new Transcript(conversationData.ConversationLog.Where(a => a.Type == ActivityTypes.Message).ToList());
+
+                var evnt = EventFactory.CreateHandoffInitiation(turnContext,
+                    new { Skill = "credit-cards" },
+                    transcript);
+
+                await turnContext.SendActivityAsync(evnt);
+            }
+            else
+            {
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                if (userText.Contains("disneyland"))
+                {
+                    await turnContext.SendActivityAsync("Sorry I cannot answer your questoin but an agent will be with you shortly");
+
+                    var channelData = @"
+  {
+    ""action"": {
+        ""name"": ""TRANSFER"",
+        ""parameters"": {
+          ""skill"": ""Fallback-Skill""
+        }
+    }
+  }";
+                    var activity = new Activity("message");
+                    activity.Text = "";
+                    activity.ChannelData = JObject.Parse(channelData);
+
+                    await turnContext.SendActivityAsync(activity);
+                    return;
+                }
+                else
+                {
+                    foreach (var country in Capitals.Keys)
+                    {
+                        if (userText.Contains(country.ToLower()))
+                        {
+                            replyText = $"The capital of {country} is {Capitals[country]}";
+                        }
+                    }
+                }
+
+                await turnContext.SendActivityAsync(MessageFactory.Text(replyText, replyText), cancellationToken);
+            }
+        }
+
+        protected override async Task OnEventAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if(turnContext.Activity.Name == "handoff.status")
```

Comment:
Consider adding a constant for this.

Reply:
Also consider extending the base class that looks at activity types and then raises events via virtual method. 

There's nothing wrong with this code, it's just a bit different than how most of our samples and docs are structured.